molecule/default/tests/test_hostname.py

```diff
@@ -0,0 +1,20 @@
+def test_hostname_set(host):
+    assert host.run("hostname -f").stdout == "dev.example.com"
+
+
+def test_hostname_resolves(host):
+    cmd = _call(host, 'dev.example.com')
+    assert cmd == '127.0.0.1'
+
+
+def test_hostname_aliases_resolve(host):
+    cmd = _call(host, 'dev')
+    assert cmd == '127.0.0.1'
+
+
+def _call(host, name):
```

Comment:
`socket.getaddrinfo('dev.example.com', 0)` should do it, but it'll need some filtering to get at the ipv4 "raw" socktype

Reply:
I believe, with testinfra, that this code runs on the local host and reaches out to execute commands on the remote host. Thus, we can't run python modules on that remote system from what I gather. That's why you use things like host.file(...) instead of using standard Python library files, etc.

Reply:
I did some testing and doc reading and you may be correct.  It seems we might be limited by the modules that testinfra provides.  

As an alternative, we could consider just removing this function, since `test_hostname_resolves()` is functionally the same in my opinion, in that it proves that the test system can resolve its own hostname.  It's likely that the `ping` and `host` commands are using the same resolver library to do the same thing.

Reply:
`test_hostname_resolves()` calls this method. This method is just used to factor out shared code between all the name aliases.

Reply:
I believe I was just referring to removal of the ping test, which you've done.  LGTM.